src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Frame.cs

```diff
@@ -145,23 +146,41 @@ public string HttpVersion
 
                 return string.Empty;
             }
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
             set
             {
-                if (value == "HTTP/1.1")
+                // GetKnownVersion returns value for "HTTP/1.1" which ReferenceEquals interned string
+                // As most common path, check for this only in fast-path and inline 
+                if (ReferenceEquals(value, "HTTP/1.1"))
                 {
                     _httpVersion = Http.HttpVersion.Http11;
                 }
-                else if (value == "HTTP/1.0")
-                {
-                    _httpVersion = Http.HttpVersion.Http10;
-                }
                 else
                 {
-                    _httpVersion = Http.HttpVersion.Unset;
+                    HttpVersionSetSlow(value);
                 }
             }
         }
 
+        [MethodImpl(MethodImplOptions.NoInlining)]
```

Comment:
Would it be that expensive to check for HTTP/1.0 in the fast path? Would inclining such a small method be such a bad thing?

I like the ReferenceEquals part of the change given string.Equals isn't inlined, but I'm not so sure the other parts of the change. I'm a little worried we're overfitting. Bug reports show HTTP/1.0 isn't that uncommon 😢 .